lib/ProductOpener/Ingredients.pm

```diff
@@ -2992,6 +3021,47 @@ sub parse_ingredients_text_service ($product_ref, $updated_product_fields_ref, $
 
 			if (not $skip_ingredient) {
 
+				# If we have a parent ingredient, check if "parent ingredient + child ingredient" is a known ingredient
+				# e.g. "vegetal oil (palm, rapeseed)" -> if we have "palm" as the child, try to transform it in "palm vegetal oil"
+
+				if (defined $parent_ref) {
+
+					# Generate the text for the canonicalized parent ingredient (so that we don't get percentages, labels etc. in it)
+					my $parent_ingredient_text
+						= display_taxonomy_tag($ingredients_lc, "ingredients", $parent_ref->{id});
+
+					my $parent_plus_child_ingredient_text;
+
+					if ($ingredients_lc eq "en") {
+						# oil (palm) -> palm oil
+						$parent_plus_child_ingredient_text = $ingredient . ' ' . $parent_ingredient_text;
+					}
+					else {
+						# huile (palme) -> huile palme
+						$parent_plus_child_ingredient_text = $parent_ingredient_text . ' ' . $ingredient;
+					}
```

Comment:
Why not test both for every language ?

Reply:
Well the more combinations we create, the greater the risk for false positives. e.g. in French, at least for oils, there's no sense in testing "[fruit / grain] huile" because it's always in the other order. We could test more combinations, but then I would prefer to wait until we have something like you suggest to see the effect of the changes.